js/dep/helpers.js

```diff
@@ -0,0 +1,74 @@
+var helpers = {
+  getUTCDatetime: function () {
+    var now = new Date()
+    var d = new Date(
+      now.getUTCFullYear(),
+      now.getUTCMonth(),
+      now.getUTCDate(),
+      now.getUTCHours(),
+      now.getUTCMinutes(),
+      now.getUTCSeconds()
+    )
+
+    var pad = function (n) {
+      return n < 10 ? '0' + n : n
+    }
+    var tz = d.getTimezoneOffset()
+    var tzs = (tz > 0 ? '-' : '+') + pad(parseInt(tz / 60, 10))
+
+    if (tz % 60 !== 0) {
+      tzs += pad(tz % 60)
+    }
+
+    if (tz === 0) {
+      tzs = 'Z'
+    }
+
+    return d.getFullYear() + '-' +
+        pad(d.getMonth() + 1) + '-' +
+        pad(d.getDate()) + 'T' +
+        pad(d.getHours()) + ':' +
+        pad(d.getMinutes()) + ':' +
+        pad(d.getSeconds()) + tzs
+  },
+  duration: function (t) {
+    var min = 60
+    var h = 60 * min
+    var hours = Math.floor(t / h)
+
+    t -= hours * h
+
+    var minutes = Math.floor(t / min)
+    t -= minutes * min
+
+    var r = ''
+    if (hours > 0) {
+      r += hours + 'h '
+    }
+    if (minutes > 0) {
+      r += minutes + 'min '
+    }
+    if (t > 0) {
+      t = t.toFixed(0)
+      r += t + 's'
+    }
+
+    if (r === '') {
+      r = '0s'
+    }
+
+    return r
+  },
+  sprintf: function (str, args) {
+    args = args || []
+    return str.replace(/(%[s])/g, function (m, i, s) {
+      var arg = args.shift()
+      if (!arg && arg !== 0) {
+        return ''
+      }
+      return arg + ''
+    })
+  }
+}
+
+module.exports = helpers
```

Comment:
It seems `helper.js` is a custom code, is there a reason why it is in the `dep` directory? It seemed a little misleading since other files in `dep` are third party files. Another drawback from that which I noticed is that it also gets ignored by eslint since `js/dep/**` is in `.eslintignore` but is that what we want? If anything, Is it possible to further break them down to `dep/vendor` (i.e for the third party code). So in `.eslintignore` we can have `js/dep/vendor/**` instead.